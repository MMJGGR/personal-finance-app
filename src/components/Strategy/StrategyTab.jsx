import React from 'react'

export default function StrategyTab() {
  return <h2 className="text-2xl font-bold text-amber-700">Strategy</h2>
}
