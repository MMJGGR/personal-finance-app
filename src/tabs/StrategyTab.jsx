import React from 'react';
export default function StrategyTab() {
  return (
    <div className="bg-white rounded-xl shadow p-4">
      <h2 className="text-2xl font-semibold mb-2 text-amber-400">Strategy</h2>
      <div className="h-40 flex items-center justify-center text-gray-400">
        [Financial Strategy Recommendations & Scenarios]
      </div>
    </div>
  );
}
