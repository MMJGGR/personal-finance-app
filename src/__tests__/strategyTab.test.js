import React from 'react'
import { render } from '@testing-library/react'
import { FinanceProvider } from '../FinanceContext'
import StrategyTab from '../components/Strategy/StrategyTab'

beforeAll(() => {
  global.ResizeObserver = class { observe() {} unobserve() {} disconnect() {} }
})

test('strategy tab placeholder snapshot', () => {
  const { container } = render(
    <FinanceProvider>
      <StrategyTab />
    </FinanceProvider>
  )
  expect(container.firstChild).toMatchSnapshot()
})
