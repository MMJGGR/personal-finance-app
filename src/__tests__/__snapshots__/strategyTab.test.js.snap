// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strategy tab placeholder snapshot 1`] = `
<div
  class="bg-white rounded-xl shadow p-6 space-y-6"
>
  <h2
    class="text-2xl font-semibold mb-4 text-amber-600"
  >
    Financial Strategy & Recommendations
  </h2>
  <section>
    <h3
      class="text-xl font-bold text-amber-800 mb-4"
    >
      Your Financial Snapshot
    </h3>
    <div
      class="grid grid-cols-1 md:grid-cols-2 gap-4"
    >
      <div>
        <p>
          <strong>
            Current Risk Profile:
          </strong>
           
           Capacity, 
           Willingness
        </p>
        <p>
          <strong>
            Calculated Risk Score:
          </strong>
           
          16
        </p>
        <p>
          <strong>
            Suggested Strategy:
          </strong>
           
          Growth
        </p>
      </div>
      <div>
        <p>
          <strong>
            Total Annual Income:
          </strong>
           
          Ksh 120,000.00
        </p>
        <p>
          <strong>
            Total Annual Expenses:
          </strong>
           
          1.03 M
        </p>
        <p>
          <strong>
            Monthly Surplus:
          </strong>
           
          -Ksh 75,500.00
        </p>
        <p>
          <strong>
            Annual Savings/Surplus:
          </strong>
           
          Ksh 0.00
        </p>
      </div>
    </div>
  </section>
  <section>
    <h3
      class="text-xl font-bold text-amber-800 mb-4"
    >
      General Recommendations
    </h3>
    <ul
      class="list-disc pl-5 space-y-2"
    >
      <li>
        Review your budget regularly to identify areas for potential savings.
      </li>
      <li>
        Consider increasing your emergency fund to cover at least 3-6 months of essential expenses.
      </li>
      <li>
        Explore investment options aligned with your risk profile and financial goals.
      </li>
      <li>
        Your current spending exceeds your income. Prioritize reducing discretionary expenses.
      </li>
    </ul>
  </section>
  <section>
    <h3
      class="text-xl font-bold text-amber-800 mb-4"
    >
      What-If Scenario Simulator
    </h3>
    <div
      class="h-48 flex items-center justify-center text-gray-400 border border-dashed rounded-md"
    >
      [Interactive charts and controls for simulating different financial scenarios]
    </div>
  </section>
</div>
`;
